### PR TITLE
Dynamic search suite changes

### PR DESCRIPTION
**Pull request by stephherbers**
## Product Description

Ticket: [USH-3666](https://dimagi-dev.atlassian.net/browse/USH-3666)
Part 1 of HQ changes at add dynamic search as a app-level config and add to query request in suite file

Part 2 HQ changes [here](https://github.com/dimagi/commcare-hq/pull/33623)
Formplayer/core changes: [here](https://github.com/dimagi/formplayer/pull/1476) & [here](https://github.com/dimagi/commcare-core/pull/1358)

review by commit

## Technical Summary

I added the flag as default to "False" for the app-level config, but really we'd ideally want this true if the FF is true, but I couldn't get the default value to be based on a FF so if you have a better way let me know. Note: from the UI perspective it is set to True, but my understanding is that would override the default value of False if and only if the UI is navigated to and saved.

Note: dynamic search should be disabled for all modules that contain autoselect due to incompatibilities of the features

## Feature Flag

DYNAMICALLY_UPDATE_SEARCH_RESULTS

### Safety story

This is behind a FF and nothing changes logically other than adding a value to the suite file

### Automated test coverage

Yes, suite coverage and translation for app UI (no other automated tests for app UI)

### QA Plan

I am not thinking of requesting QA

### Rollback instructions

- [X] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [X] Risk label is set correctly
- [X] The set of people pinged as reviewers is appropriate for the level of risk of the change

[USH-3666]: https://dimagi-dev.atlassian.net/browse/USH-3666?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: corehq/apps/app_manager/static/app_manager/json/commcare-app-settings.yml [hunk 1]
```diff
@@ -169,3 +169,12 @@
     - ['only_flat_fixture', 'Only Flat Fixture']
     - ['only_hierarchical_fixture', 'Only Hierarchical Fixture']
   default: 'project_default'
+
+
+- id: split_screen_dynamic_search
+  name: Dynamic Search for Split Screen Case Search
+  description: Enables searching as search values change after initial search for Split Screen Case Search
```
Comment (nospame) [inline corehq/apps/app_manager/static/app_manager/json/commcare-app-settings.yml:176]: nit: I think I dislike how many times "search" appears in this sentence. It also looks like the convention, if we have one, would be to use "Enable" without the s.  Maybe something like "Enable searching as input values change after initial Split Screen Case Search"

**Review by nospame — COMMENTED** [review]
This looks good to me as long as it's coordinated timing-wise with the Formplayer/core changes (those have to come first, right?) Also still see
```
 FAIL: corehq.apps.app_manager.tests.test_suite_inline_search:InlineSearchShadowModuleTest.test_suite
```
despite changes to test XML, but would feel fine approving once it's addressed.


> ideally want this true if the FF is true, but I couldn't get the default value to be based on a FF so if you have a better way let me know.

I assume this would probably need a data migration.

**Comment by stephherbers** [comment]
> > ideally want this true if the FF is true, but I couldn't get the default value to be based on a FF so if you have a better way let me know.
> 
> I assume this would probably need a data migration.

I think I'm more talking about after the fact that's it's implemented, because with the logic right now, once someone enabled the FF on the environment the default value would still be false when ideally at all times the FF is turned on this would switch. Does that make sense?

**Comment by orangejenny** [comment]
Code looks good overall.

> I added the flag as default to "False" for the app-level config, but really we'd ideally want this true if the FF is true, but I couldn't get the default value to be based on a FF so if you have a better way let me know. 

I don't think this is possible, since the user/domain isn't available when the default attribute value is set. (We do have code to get the current global request, but this doesn't seem like a strong enough reason to go mucking around in global-level code.)

> Note: from the UI perspective it is set to True, but my understanding is that would override the default value of False if and only if the UI is navigated to and saved.

Can you update that? The UI should match the true value. I think changing the value in the yaml file will do it.

Is there test coverage for the new behavior? I was expecting to see `dynamic_search="true"` in test code somewhere.

**Comment by stephherbers** [comment]
> Note: from the UI perspective it is set to True, but my understanding is that would override the default value of False if and only if the UI is navigated to and saved.
> 
> Can you update that? The UI should match the true value. I think changing the value in the yaml file will do it.

Updated [81542a2](https://github.com/dimagi/commcare-hq/pull/33622/commits/81542a2095e23a1fff56eff2d2b512724b394f63)

**Review by orangejenny — APPROVED** [review]

**Comment by nospame** [comment]
> 
> I think I'm more talking about after the fact that's it's implemented, because with the logic right now, once someone enabled the FF on the environment the default value would still be false when ideally at all times the FF is turned on this would switch. Does that make sense?

Ahh I get it now, that makes sense.

**Review by nospame — APPROVED** [review]

**Comment by stephherbers** [comment]
@nospame @orangejenny changes made since last approval. There are still the 2 failing lints, but not sure how to fix (new line causes test to fail) so if you have any thoughts let me know

File: corehq/apps/app_manager/static/app_manager/json/commcare-settings-layout.yml [hunk 2]
```diff
@@ -61,6 +61,7 @@
     - hq.translation_strategy
     - hq.mobile_ucr_restore_version
     - hq.location_fixture_restore
+    - hq.split_screen_dynamic_search
```
Comment (orangejenny) [inline corehq/apps/app_manager/static/app_manager/json/commcare-settings-layout.yml:64]: This reminds me we might want to split out a "web apps" section and/or rename the android section to something like "UI", but that's not part of this PR.

**Review by orangejenny — APPROVED** [review]

**Comment by stephherbers** [comment]
formplayer and core merged

**Pull request merged**